### PR TITLE
Export students as CSV

### PR DESCRIPTION
**Pull request by hisahi**
### Short description
#611: Export student table CSV as instructor

### DoD
I have:
- [x] added actual code.
- [x] produced clean code that passes ESLint.
- [x] code that actually does what it should.
- [x] documented the code or added documentation to the wiki.
- [x] added or modified test(s).
- [x] test(s) that actually pass.

### CODE REVIEW

File: labtool2.0/src/components/pages/CoursePage/TeacherMain.js [hunk 1]
```diff
@@ -53,6 +54,9 @@ export const CoursePageTeacherMain = props => {
           <Button size="small">Send email to multiple students</Button>
         </Link>
       }
+      <Button size="small" onClick={() => exportCSV()}>
```
Comment (mjaakko) [inline labtool2.0/src/components/pages/CoursePage/TeacherMain.js]:
```suggestion
      <Button size="small" onClick={exportCSV}>
```

File: labtool2.0/src/components/pages/CoursePage.js [hunk 2]
```diff
@@ -59,6 +59,80 @@ export const CoursePage = props => {
...
+      for (var j = 0; j < student.weeks.length; j++) {
+        if (student.weeks[j].weekNumber === selectedInstance.weekAmount + 1) {
+          finalPoints = student.weeks[j].points
+        } else if (student.weeks[j].weekNumber) {
+          weekPoints[student.weeks[j].weekNumber] = student.weeks[j].points
+        }
+      }
+      for (let i = 1; i <= props.selectedInstance.weekAmount; ++i) {
+        const points = weekPoints[i]
+        values.push(points || points === 0 ? points : '-')
```
Comment (mjaakko) [inline labtool2.0/src/components/pages/CoursePage.js]: I think these CSV files are intended to be machine-readable and in that case `-` would not be a good value for missing points.

Comment (hisahi) [reply labtool2.0/src/components/pages/CoursePage.js]: It's not certain whether this is for being machine-readable or just for importing into a spreadsheet.

**Comment by hisahi** [comment]
> Also, maybe it would be possible to create a generic "Export CSV" component that would be easier to unit test?

It's unlikely this would be used more than once, so it's better to somehow write tests for the existing code - not sure what the best approach is, though. Mocking downloadFile?

**Comment by mjaakko** [comment]
> > Also, maybe it would be possible to create a generic "Export CSV" component that would be easier to unit test?
> 
> It's unlikely this would be used more than once, so it's better to somehow write tests for the existing code - not sure what the best approach is, though. Mocking downloadFile?

Mocking `downloadFile` would be fine. We want to test that exported CSV is in correct format and has correct data in all columns.

**Pull request merged**